src/test/java/org/reaktivity/k3po/nukleus/ext/DuplexIT.java

```diff
@@ -318,6 +318,66 @@ public void shouldThrottleInitialOnlyServerSentData() throws Exception
 
     @Test
     @Specification({
+        "server.sent.throttle.stream/client",
+        "server.sent.throttle.stream/server"
+    })
+    public void shouldThrottleClientSentDataPerStream() throws Exception
+    {
+        k3po.finish();
+    }
+
+    // should pass because even if throttling is disabled, the receiver window is big enough to receive the message
+    @Test
+    @Specification({
+        "client.sent.throttle.off/client",
+        "client.sent.throttle.off/server"
+    })
+    public void shouldReceiveMessageThrottleOff() throws Exception
+    {
+        k3po.finish();
+    }
+
+    // should fail because the receiver window is not big enough and window adjustment is disabled
+    @Test
+    @Specification({
+        "client.sent.throttle.off.negative/client",
+        "client.sent.throttle.off.negative/server"
+    })
+    public void shouldNotReceiveMessageThrottleOff() throws Exception
+    {
+        thrown.expect(anyOf(isA(ComparisonFailure.class),
+                hasProperty("failures", hasItem(isA(ComparisonFailure.class)))));
+        k3po.finish();
+    }
+
+    // should fail (please see readme)
+    @Test
+    @Specification({
+        "server.set.window0.throttle.on/client",
+        "server.set.window0.throttle.on/server"
+    })
+    public void shouldReceiveMessageWindow0ThrottleingOn() throws Exception
```

Comment:
I think this test will fail due to a timeout. That means it will take 10 seconds, and that's multiplied by three including the other tests (SimplexIT and HalfDuplexIT) so will significantly increase the build time. The purpose of the test I think is to verify that the writer does not violate the window (i.e. write data larger than the window). That is already amply verified by most of the other tests: they would all fail if any write violated the window, because the other end would send a reset (and that is tested by server.sent.overflow/ and client.sent.overflow).

So I think we should remove this test too.

Reply:
Done.